src/Microsoft.Management.Deployment/CatalogPackage.cpp

```diff
@@ -126,15 +126,27 @@ namespace winrt::Microsoft::Management::Deployment::implementation
                     auto latestVersionImpl = winrt::make_self<wil::details::module_count_wrapper<
                         winrt::Microsoft::Management::Deployment::implementation::PackageVersionInfo>>();
                     latestVersionImpl->Initialize(std::move(latestApplicableVersion));
-                    m_defaultInstallVersion = *latestVersionImpl;
+                    m_latestApplicableVersion = *latestVersionImpl;
                 }
             });
     }
 
     Microsoft::Management::Deployment::PackageVersionInfo CatalogPackage::DefaultInstallVersion()
     {
-        InitializeDefaultInstallVersion();
-        return m_defaultInstallVersion;
+        InitializeLatestApplicableVersion();
+
+        if (m_latestApplicableVersion)
+        {
+            return m_latestApplicableVersion;
+        }
+        else if (AvailableVersions().Size() > 0)
+        {
+            return GetPackageVersionInfo(AvailableVersions().GetAt(0));

```

Comment:
I would rather that this decision be made in the `call_once`, but I also refactored this entire function elsewhere in my upcoming PR.  So I will move that when I handle the merge conflict that this creates...